src/controllers/threadController.js

```diff
@@ -69,6 +69,21 @@ export const updateThread = async (req, res) => {
   }
 };
 
+export const favoriteThread = async (req, res) => {
+  try {
+    console.log('Server::favorite a thread - running favoriteThread');
+    const id = req.params.id;
+    const userId = req.body.poster;
```

Comment:
why is the userId is the poster? are u adding a thread to its own author's favorites? I think u need to get the user id from cookie actually.

Reply:
Oh!! that's right. I will figure out this part